app/src/main/java/com/infomaniak/mail/ui/newMessage/selectMailbox/SelectMailboxViewModel.kt

```diff
@@ -111,4 +127,10 @@ class SelectMailboxViewModel @Inject constructor(
         val avatarUrl: String?,
         val initials: String,
     )
+
+    sealed interface UiState {
+        data object Loading : UiState
+        data class DefaultMailbox(val defaultMailbox: SelectedMailboxUi) : UiState
+        data class SelectMailbox(val selectedMailbox: State<SelectedMailboxUi?>) : UiState
```

Comment:
The goal of having a UiState like this is that you can have a single state in your ui. Here `selectedMailbox` can be simplified by creating another UiState instead of having to bother with a `State<SelectedMailboxUi?>`. This will help you avoid having to deal with two different states and deal will nullable values